go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -284,6 +243,33 @@ func (t *TxThrottler) Throttle() (result bool) {
 	return result
 }
 
+// txThrottlerConfig holds the parameters that need to be
+// passed when constructing a TxThrottler object.
+type txThrottlerConfig struct {
+	// enabled is true if the transaction throttler is enabled. All methods
+	// of a disabled transaction throttler do nothing and Throttle() always
+	// returns false.
+	enabled bool
+
+	topoServer      *topo.Server
+	throttlerConfig *throttlerdatapb.Configuration
+	// healthCheckCells stores the cell names in which running vttablets will be monitored for
+	// replication lag.
+	healthCheckCells []string
+}
+
+// txThrottlerState holds the state of an open TxThrottler object.
+type txThrottlerState struct {
+	// throttleMu serializes calls to throttler.Throttler.Throttle(threadId).
+	// That method is required to be called in serial for each threadId.
+	throttleMu      sync.Mutex
+	throttler       ThrottlerInterface
+	stopHealthCheck context.CancelFunc
+
+	healthCheck      discovery.HealthCheck
+	topologyWatchers []TopologyWatcherInterface
+}
+
```

Comment:
Can you please confirm this is just shuffling of code? It's hard to see if anything was actually changed.

Reply:
@shlomi-noach yes, this is a copy/paste change of existing structs 👍.

Related to readability, I could use some feedback: I moved the `txThrottlerState` and `txThrottlerConfig` structs further-down in the file to where they're most relevant, however `state *txThrottlerState` is also part of the `TxThrottler` struct near the start of the file. So I had a question, would you prefer:
1. `txThrottlerConfig` and `txThrottlerState` are moved further down, near it's methods
2. `txThrottlerConfig` and `txThrotterState` structs are defined before `TxThrottler`, which contains `state *txThrottlerState`
    - Drawback: the struct methods are near the end of the file, not the start

🙇

Reply:
I would move the structs up to the beginning AND reorder the struct methods so that `state` and `config` methods are defined right after the struct definition of `TxThrottler`. Such wholesale  reorganization will definitely make the PR harder to review, so you might want to get the panic changes all reviewed and squared away and then do the re-org as a separate commit.

Reply:
Agree with @deepthi . Also, when there's too many structs to fit in one file I personally prefer to start splitting into multiple files.